tests/beaker_tests/cisco_bfd_global/test_bfd_global.rb

```diff
@@ -0,0 +1,281 @@
+###############################################################################
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  master:           master,
+  agent:            agent,
+  operating_system: 'nexus',
+  platform:         'n(7|9)k',
+  resource_name:    'cisco_bfd_global',
+}
+
+default_interval_plat_1 = %w(250 250 3)
+default_interval_plat_2 = %w(50 50 3)
+non_default_interval = %w(100 100 25)
+non_default_ipv4_interval = %w(200 200 50)
+non_default_ipv6_interval = %w(500 500 30)
+non_default_fabricpath_interval = %w(750 350 35)
+
+# Test hash test cases
+tests[:default_plat_1] = {
+  desc:           '1.1 Defaults for n3k',
+  platform:       'n3k',
+  title_pattern:  'default',
+  preclean:       'cisco_bfd_global',
+  manifest_props: {
+    echo_interface:        'default',
+    echo_rx_interval:      'default',
+    interval:              'default',
+    ipv4_echo_rx_interval: 'default',
+    ipv4_interval:         'default',
+    ipv4_slow_timer:       'default',
+    ipv6_echo_rx_interval: 'default',
+    ipv6_interval:         'default',
+    ipv6_slow_timer:       'default',
+    slow_timer:            'default',
+    startup_timer:         'default',
+  },
+  code:           [0, 2],
+  resource:       {
+    echo_interface:        'false',
+    echo_rx_interval:      250,
+    interval:              default_interval_plat_1,
+    ipv4_echo_rx_interval: 250,
+    ipv4_interval:         default_interval_plat_1,
+    ipv4_slow_timer:       2000,
+    ipv6_echo_rx_interval: 250,
+    ipv6_interval:         default_interval_plat_1,
+    ipv6_slow_timer:       2000,
+    slow_timer:            2000,
+    startup_timer:         5,
+  },
+}
+
```

Comment:
Having separate tests for each platform is certainly one way to do this but I think it would be cleaner to have one test, where you define the inputs separately from the test hash. Also, we have prior artifacts for handling unsupported properties on a per platform basis.

For example:

```

tests[:default] = {
  desc:           '1.1 Default Properties',
  title_pattern:  'default',
  preclean:       'cisco_bfd_global',
  manifest_props: {
    echo_interface:        'default',
    echo_rx_interval:      'default',
    interval:              'default',
    ipv4_echo_rx_interval: 'default',
    ipv4_interval:         'default',
    ipv4_slow_timer:       'default',
    ipv6_echo_rx_interval: 'default',
    ipv6_interval:         'default',
    ipv6_slow_timer:       'default',
    slow_timer:            'default',
    startup_timer:         'default',
  },
  resource:      {
    echo_interface:        'false',
    echo_rx_interval:      50,
    ipv4_echo_rx_interval: 50,
    ipv4_slow_timer:       2000,
    ipv6_echo_rx_interval: 50,
    ipv6_slow_timer:       2000,
    slow_timer:            2000,
    startup_timer:         5,
  }
}

# Per platform default values
resource = {
  n3k: {
    echo_rx_interval:      250,
    interval:              default_interval_plat_1,
    ipv4_echo_rx_interval: 250,
    ipv4_interval:         default_interval_plat_1,
    ipv6_echo_rx_interval: 250,
    ipv6_interval:         default_interval_plat_1,
  },
  n9k: {
    ipv4_interval:         default_interval_plat_2,
    ipv6_interval:         default_interval_plat_2,
  },
}

case platform
when /n3k/
  tests[:default][:resource].merge!(resource[:n3k])
when /n(8|9)k/
  tests[:default][:resource].merge!(resource[:n9k])
end

def unsupported_properties(tests, id)
  unprops = []
  if platform[/n(5|6)k/]
    unprops <<
      :ipv4_interval <<
      :ipv6_interval
  end
  unprops
end
```